control/control_loop.go

```diff
@@ -61,12 +62,13 @@ func createLoop(logger logging.Logger, cfg Config, m Controllable) (*Loop, error
 		return nil, errors.New("loop frequency shouldn't be 0 or above 200Hz")
 	}
 	l.dt = time.Duration(float64(time.Second) * (1.0 / (l.cfg.Frequency)))
-	for _, bcfg := range cfg.Blocks {
+	for i, bcfg := range cfg.Blocks {
 		blk, err := createBlock(bcfg, logger)
 		if err != nil {
 			return nil, err
 		}
 		l.blocks[bcfg.Name] = &controlBlockInternal{blk: blk, blockType: bcfg.Type}
+		l.blocks[bcfg.Name].blockIndex = i
```

Comment:
[q] Why are you adding the block index?

Reply:
also don't think this is necessary any longer with new changes